src/Extractor/ContentExtractor.php

```diff
@@ -669,4 +686,51 @@ private function extractBody($detectBody, $xpathExpression, \DOMNode $node, $typ
 
         return false;
     }
+
+    /**
+     * Extract and apply a callback to an entity according to a pattern
+     *
+     * @param string $entity            Entity to look for
+     * @param string $pattern           Pattern to look for
+     * @param callable $returnCallback  Function to apply on the value
+     *
+     * @return bool Telling if the entity has been found
+     */
+    private function extractEntityFromPattern($entity, $pattern, $returnCallback = null)
+    {
+        // we define the default callback here
+        if (!is_callable($returnCallback)) {
+            $returnCallback = function($e) {
+                return $e;
+            };
+        }
+
+        $elems = $this->xpath->evaluate($pattern, $this->readability->dom);
+        $entityValue = null;
+
+        if (is_string($elems)) {
+            $entityValue = $returnCallback($elems);
+            $this->logger->log('debug', "{$entity} expression evaluated as string: {{$entity}}", array($entity => $entityValue));
+            $this->logger->log('debug', '...XPath match: {pattern}', array('pattern', $pattern));
+        } elseif ($elems instanceof \DOMNodeList && $elems->length > 0) {
+            $entityValue = $returnCallback($elems->item(0)->textContent);
+            $this->logger->log('debug', "{$entity} matched: {{$entity}}", array($entity => $entityValue));
+            $this->logger->log('debug', '...XPath match: {pattern}', array('pattern', $pattern));
+
+            // remove entity from document
+            try {
+                $elems->item(0)->parentNode->removeChild($elems->item(0));
+            } catch (\DOMException $e) {
+                // do nothing
+            }
+        }
+
+        if ($entityValue !== null)
+        {
+            $this->{$entity} = $entityValue;
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
This `else` isn't necessary since you already exit from the function in the below `if`